langchain/src/experimental/masking/parser.ts

```diff
@@ -0,0 +1,139 @@
+import { MaskingTransformer } from "./transformer.js";
+import { MaskingParserConfig } from "./types.js";
+
+/**
+ * MaskingParser class for handling the masking and rehydrating of messages.
+ */
+export class MaskingParser {
+  private transformers: MaskingTransformer[];
+  private state: Map<string, string>;
+  private config: MaskingParserConfig;
+
+  constructor(config: MaskingParserConfig = {}) {
+    this.transformers = config.transformers || [];
+    this.state = new Map<string, string>();
+    this.config = config;
+  }
+
+  /**
+   * Adds a transformer to the parser.
+   * @param transformer - An instance of a class extending MaskingTransformer.
+   */
+  addTransformer(transformer: MaskingTransformer) {
+    this.transformers.push(transformer);
+  }
+
+  /**
+   * Getter method for retrieving the current state.
+   * @returns The current state map.
+   */
+  public getState(): Map<string, string> {
+    return this.state;
```

Comment:
Is this necessary?

Reply:
This is necessary to consume and rehydrate state asynchronously. Imagine lambda one masks a string -> mask state stored in kv -> lambda x needs the state to rehydrate the masked values.

Reply:
Right, but this method vs. just having a public property?